Documentation/GettingStarted.md

```diff
@@ -0,0 +1,371 @@
+
+Our app will be leverage a simple API that takes a search query and returns a JSON response of results from the iTunes store. The API takes the form: `https://itunes.apple.com/search?term=siracusa&limit=100`
+
+Pilot does not dictate how your application fetches from the network or stores data locally (although making that easier is on the roadmap). For the purposes of this example, we'll use a very basic [SearchService](../Examples/iTunesSearch/Shared/SearchService.swift) class to take a query string, and item fetch limit to fetch our JSON from the above API.
+
+```swift
+class SearchService {
+	func search(
+		term: String,
+		limit: Int,
+		completion: @escaping ([Media]?, ServiceError?) -> Void
+	) {
+		// ...
+	}
+```
+
+Don't worry about why it's returning `[Media]` just yet, we'll cover how the JSON gets converted shortly. And yes, there are better signatures for that completion block, but we're keeping things simple.
+
+Now that we have a basic search API, we want to start piping that data into Pilot. To do so, we start with our first core protocol, `Model`.
+
+## `Model`
+
+In Pilot, a [`Model`](../Core/Source/MVVM/Model.swift) is a protocol that represents a **value type of pure stateless data**. For nearly all applications, this represents "one of the things in your scrolling list" — in this case, a result from an iTunes Search. In your application, it may be a message in a group chat, a post in a news feed, a comment in a thread, or a task in your todo app.
+
+In our application, the JSON from iTunes returns a list of model objects. Here is a song:
+
+```json
+{
+	"kind": "song",
+	"artistId": 5448756,
+	"collectionId": 719245563,
+	"trackId": 719245998,
+	"artistName": "The Long Winters",
+	"collectionName": "Ultimatum - EP",
+	"trackName": "The Commander Thinks Aloud",
+	"previewUrl": "http://audio.itunes.apple.com/apple-assets-us-std-000001/<snipped>/.m4a",
+	"artworkUrl100": "http://is2.mzstatic.com/image/thumb/<snipped>/100x100bb.jpg",
+	"trackTimeMillis": 326053,
+	"trackNumber": 1,
+	"trackCount": 6,
+	// ...
+}
+```
+
+Representing this data in Swift would look like this:
+
+```swift
+struct Song {
+	var artistId: Int
+	var collectionId: Int
+	var trackId: Int
+	var artistName: String
+	var collectionName: String
+	var trackName: String
+	var preview: URL
+	var artwork: URL?
+	var durationMilliseconds: Int
+	var trackNumber: Int?
+	var trackCount: Int?
+}
+```
+
+Your model object doesn't have to come from JSON, it could come from Core Data, SQLite, or any other source. 
+
+> How to serialize & deserialize from your data source into a model struct like the above example is beyond the scope of this document.
+
+> The example application happens to add an initializer to the `Song` struct to inflate from JSON, along with a protocol to make that common across the other model types returned from iTunes. If interested, you can see that [here](../Examples/iTunesSearch/Shared/ModelSerialization.swift).
```

Comment:
We could actually replace this with `codable` now to be a little more idiomatic in swift 4

Reply:
Yes! agreed. I added a TODO.